app/controllers/all_search_result_ids_controller.rb

```diff
@@ -0,0 +1,72 @@
+# This controller (just used for #add_to_cart) runs a Blacklight search for all works and collections specified in the params.
+# and returns JUST the friendlier ids in the search results, in a JSON array.
+class AllSearchResultIdsController < CatalogController
+  before_action :authenticate_user! # need to be logged in
+
+  configure_blacklight do |config|
+    config.search_state_fields << :all_search_result_ids
+  end
+
+
+  # A CartItem doesn't contain a lot of data, so this goes fast, even for thousands of ids
+  def add_to_cart
+    conn = ActiveRecord::Base.connection
+    CartItem.transaction do
+      all_ids_from_search.each_slice(500) do |ids|
+        conn.execute add_friendlier_ids_to_cart_sql(ids)
```

Comment:
I still think there oughta be a way to do this with ActiveRecord insert_all and/or upsert_all that would be just as performant, and somewhat neater, without having to maintain our own SQL. 

But there's really nothing _wrong_ with doing it this way, and you have done it very tidily!  Not sure if you experimented with insert_all/upsert_all.    I wonder if we should try it -- are your tests good enough that if they pass we are confident? :). 

But also really nothing wrong with this way.